app/controllers/employment/AddEmploymentController.scala

```diff
@@ -44,14 +44,14 @@ class AddEmploymentController @Inject()(implicit val cc: MessagesControllerCompo
                                         ec: ExecutionContext
                                        ) extends FrontendController(cc) with I18nSupport with SessionHelper {
 
-  def show(taxYear: Int): Action[AnyContent] = authAction.async { implicit user =>
+  def show(taxYear: Int): Action[AnyContent] = (authorisedAction andThen TaxYearAction.taxYearAction(taxYear)).async { implicit user =>
```

Comment:
I would probably create a method that would return the composed action so that we do not have to do this code here. Potential method name is authorisedTazYearAction. You can put it in a common place (an object). If you do that you will not need to inject AuthorisedAction and you can use it directly similar to the TaxYearAction. The same comment applies to all the controllers you have updated.

Further, my understanding is that the new action needs to be used in all the controllers. I think you have missed to add it to quite a few of the controllers.

Reply:
Updated as discussed yesterday